orchagent/pfcwdorch.cpp

```diff
@@ -312,51 +344,50 @@ PfcWdSwOrch<DropHandler, ForwardHandler>::PfcWdSwOrch(
         DBConnector *db,
         vector<string> &tableNames,
         vector<sai_port_stat_t> portStatIds,
-        vector<sai_queue_stat_t> queueStatIds):
+        vector<sai_queue_stat_t> queueStatIds,
+        vector<sai_queue_attr_t> queueAttrIds):
     PfcWdOrch<DropHandler,
     ForwardHandler>(db, tableNames),
     m_pfcWdDb(new DBConnector(PFC_WD_DB, DBConnector::DEFAULT_UNIXSOCKET, 0)),
     m_pfcWdTable(new ProducerStateTable(m_pfcWdDb.get(), PFC_WD_STATE_TABLE)),
     c_portStatIds(portStatIds),
-    c_queueStatIds(queueStatIds)
+    c_queueStatIds(queueStatIds),
+    c_queueAttrIds(queueAttrIds)
 {
     SWSS_LOG_ENTER();
 
     string platform = getenv("platform") ? getenv("platform") : "";
 
-    if (platform == "")
+    if (platform == MLNX_PLATFORM_SUBSTRING)
     {
-        SWSS_LOG_ERROR("Platform environment variable is not defined");
-        return;
+        SWSS_LOG_NOTICE("Current platform is mlnx");
+    }
+    else
+    {
+        platform = BROADCOM_PLATFORM;
+        SWSS_LOG_NOTICE("Current platform is broadcom");
     }
 
     string detectSha, restoreSha;
     string detectPluginName = "pfc_detect_" + platform + ".lua";
     string restorePluginName = "pfc_restore_" + platform + ".lua";
 
-    try
```

Comment:
Do you want not to handle it?
If we'll fail to load .lua script due to, for example, syntax error orchagent will crash.

Reply:
I agree it might be better to catch the exception to prevent orchagent from crashing. Update the change.